### ISSUE TITLE
If possible please do not assume /usr/lib/R/share/make/shlib.mk

### ISSUE THREAD

**Issue by eddelbuettel** [mjwoods/RNetCDF#130]
On Debian / Ubuntu systems since time immortal:

```sh
$ ls -l /usr/lib/R/share/make/shlib.mk
ls: cannot access '/usr/lib/R/share/make/shlib.mk': No such file or directory
$ ls -l /usr/share/R/share/make/shlib.mk
-rw-r--r-- 1 root root 661 Aug  5 16:56 /usr/share/R/share/make/shlib.mk
$ 
```

so the build breaks.  You can make use of `R.home("share")` to get to it portably:

```r
> R.home("share")
[1] "/usr/share/R/share"
> file.path(R.home("share"), "make", "shlib.mk")
[1] "/usr/share/R/share/make/shlib.mk"
> file.info(file.path(R.home("share"), "make", "shlib.mk"))
                                 size isdir mode               mtime               ctime               atime uid gid uname grname
/usr/share/R/share/make/shlib.mk  661 FALSE  644 2023-08-05 16:56:56 2023-08-05 22:05:45 2023-10-09 11:15:37   0   0  root   root
> 
```

**Comment by eddelbuettel** [comment]
Related to at least #128

**Comment by vlulla** [comment]
OOPS...should've looked at the pull request. Anyways, thanks @eddelbuettel for sharing a portable way to do this!

**Comment by mjwoods** [comment]
Thanks for reporting the problem @eddelbuettel . This is quite unexpected, because I have a set of tests running with Github actions using Ubuntu images. Thanks for suggesting a solution in your PR, and I will merge it after a few more tests of my own. Unfortunately, it may take a week or so before I can update the package on CRAN.

**Comment by eddelbuettel** [comment]
Ubuntu != Ubuntu.  I run with the binary that is the official build, derived from the Debian package, and also provided as backports at CRAN in the [bin/linux/ubuntu directory](https://cran.r-project.org/bin/linux/ubuntu).  I run those in my CI too.  If you use a CI setup provided by others it may use a different R binary even when on Ubuntu.  

In any event, you also get reports from an Arch user.  In short `R.home("shared")` is the more portable approach -- one of those things.  If CRAN (which uses their local builds, also without his issue) doesn't bug you a week should be perfectly fine.

**Comment by mjwoods** [comment]
Resolved by #128

**Comment by tillea** [comment]
Hi,
it would be great if you could tag a new minor (micro) release.  This would help me to update the Debian package without adding the according patches.
Thanks a lot, Andreas.

**Comment by mjwoods** [comment]
Hi @tillea, I am happy to create a micro release for you tomorrow morning (Australian time). If you decide to patch the Debian package before then, please let me know. Also, I should be ready to release a new minor version this weekend or early next week, and it would avoid a micro release if you could delay until then.

**Comment by tillea** [comment]
Am Tue, Oct 17, 2023 at 12:48:57PM -0700 schrieb Milton Woods:
> OK then, I will skip the micro release and focus on getting out the minor release ASAP. That will fix the Debian & Ubuntu packages as well as addressing further problems reported by CRAN.
Sounds good, Andreas.

**Comment by ved-ux** [comment]
I am getting this error as shown below by entering install.packages("RNetCDF") in r studio 

** libs
WINDOWS="FALSE" \
        MAKE="make" \
        R_HOME="/usr/lib/R" \
        R_ARCH="" \
        /bin/bash ../tools/make-recursive.sh
make -f /usr/lib/R/etc//Makeconf -f /usr/lib/R/share/make/shlib.mk -f Makefile.common OBJECTS= attribute.o common.o convert.o dataset.o dimension.o group.o init.o type.o udunits.o variable.o SHLIB=RNetCDF$(SHLIB_EXT) RNetCDF_all
make[1]: Entering directory '/tmp/RtmpIXNCkh/R.INSTALLa632040629d4a/RNetCDF/src'
make[1]: /usr/lib/R/share/make/shlib.mk: No such file or directory
make[1]: Leaving directory '/tmp/RtmpIXNCkh/R.INSTALLa632040629d4a/RNetCDF/src'
make[1]: *** No rule to make target '/usr/lib/R/share/make/shlib.mk'.  Stop.
make: *** [Makefile:14: all] Error 2
ERROR: compilation failed for package ‘RNetCDF’
* removing ‘/home/wrf_hydro_fnl/R/x86_64-pc-linux-gnu-library/4.3/RNetCDF’
Warning in install.packages :
  installation of package ‘RNetCDF’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpjUAnZP/downloaded_packages’

**Comment by mjwoods** [comment]
Hi all, I have uploaded a new release (RNetCDF_2.8-1) to CRAN, which should be available for downloads over the next few days. The new version fixes this issue (#130) and several additional issues found by CRAN in RNetCDF_2.7-1. 

Thanks for reporting this issue and for your patience while I investigated. To avoid similar issues in future, I have added tests using R from the Ubuntu r-base-dev package to my Github actions.
